src/astro_image_display_api/interface_definition.py

```diff
@@ -52,41 +52,19 @@ class ImageViewerInterface(Protocol):
 
     # The methods, grouped loosely by purpose
 
-    # Methods for loading data
+    # Method for loading data
     @abstractmethod
-    def load_fits(self, file: str | os.PathLike) -> None:
+    def load(self, data: Any) -> None:
         """
-        Load a FITS file into the viewer.
+        Load data into the viewer. At a minimum, this should allow a FITS file
```

Comment:
Depending on how the Jdaviz discussions go, we could also make this more flexible, like "if you already load data into the app another way, this method would add that data to the viewer this method is attached to" or something like that.

Reply:
I am thinking more like, people would still use `jdaviz.load(...)` outside of this API scope to read data to the app. Then then `jdaviz.viewers["blah"].load(data_label)` to put the data into the viewer with this API. So this means:

* Jdaviz does not need to worry about fitting this API at app level and limit its scope to viewer level.
* This API in Jdaviz only takes a string and would emit exception to tell user to use app level load otherwise.

Of course the exception part is not very elegant. You can also have `viewer.load` try to be smart and call app load, as you said above, but it increases the complexity of the method.

Reply:
A more controversial suggestion is that we recognize that different backends might have very different ways to load data, and actually drop `load` from the API itself. This means the API here would just assume the data is already available. 🔥